geofetch/geofetch.py

```
@@ -1789,15 +1796,22 @@ def _download_processed_file(self, file_url: str, data_folder: str) -> bool:
                 return True
 
             except IOError as e:
-                _LOGGER.error(str(e))
-                # The server times out if we are hitting it too frequently,
-                # so we should sleep a bit to reduce frequency
-                sleeptime = (ntry + 1) ** 3
-                _LOGGER.info(f"Sleeping for {sleeptime} seconds")
-                time.sleep(sleeptime)
-                ntry += 1
-                if ntry > 4:
-                    raise e
+                if os.name == "nt":
+                    _LOGGER.error(f"{e}")
+                    raise OSError(
```

Should we raise an error here? I feel like we can just check for the `wget` in the beginning and handle it from there, no?

---

Same, here..

---

Could you try to execute `wget --version` at the start and mark it unavailable if it raises an error?
In Powershell `wget` is aliased to `Invoke-WebRequest` so if you type `wget`, something will happen but it's not really wget (different command line args etc.) I don't know if Python will cause Powershell commands to be executed or not, but if it does, that could also cause confusion.

---

Do you know if any of this is a problem when using WSL? I thought WSL leveled the playing field when it came to posix compatibility with windows. Unfortunately I don't have immediate access to a Windows machine to test some of these things.

But I kind of agree with @pedro-w in that it should be pretty easy to do an initial check at the beginning to see if `wget` is available at all and then raise errors/move forward from there

---

It does appear to work for me under WSL but I only ran one command and saw that it started downloading files ([comment](https://github.com/pepkit/geofetch/issues/130#issuecomment-1888689387))
... but WSL could be one of a number of distros and might not have wget installed (same as Linux, actually)